packages/coffeescript/plugin/compile-coffeescript.js

```diff
@@ -83,8 +83,8 @@ export class CoffeeCompiler extends CachingCompiler {
 
     let sourceMap = JSON.parse(output.v3SourceMap);
 
-    if (source.match(/(`|import|export|function\*)/)) {
-      // If source contains backticks or ES2015+ features,
+    if (source.match(/`|\bimport\b|\bexport\b|\byield\b/)) {
```

Comment:
The following should have a noticeable performance increase over this, especially on larger files:
```js
source.match(/`|\b(?:import|export|yield)\b/)
```

********
And _if_ attempting to eliminate some false-positives, perhaps:

```js
source.match(/`|(?:;|^)\h*\b(?:import|export|yield)\b/)
```
Though I'd venture to say that it might be possible to break this logic – haven't given it a lot of thought.  My assumption here is that `import`/`export`/`yield` are either at the beginning of a line or after a semi-colon, proceeded by optional whitespace.

Again, any benchmarks you could run here might be nice to see.  Setting the env. variable `METEOR_PROFILE=50` might be your friend in seeing the differences in compilation time.